### PR TITLE
Add a Roadrunner integration

### PR DESCRIPTION
**Pull request by bwoebi**
### Description

Adds a small integration for handling Roadrunner PSR7 requests and responses.

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [x] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: src/Integrations/Integrations/Roadrunner/RoadrunnerIntegration.php [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    public function init()
+    {
+        if (!self::shouldLoad(self::NAME)) {
+            return Integration::NOT_LOADED;
+        }
+
+        $integration = $this;
+
+        ini_set("datadog.trace.auto_flush_enabled", 1);
+        ini_set("datadog.trace.generate_root_span", 0);
```
Comment (morrisonlevi) [inline src/Integrations/Integrations/Roadrunner/RoadrunnerIntegration.php:45]: Won't it already be too late for `generate_root_span` to be set?

File: src/Integrations/Integrations/Roadrunner/RoadrunnerIntegration.php [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+            },
+            'posthook' => function ($psr, $scope, $args, $retval, $exception) use (&$activeSpan, $integration) {
+                if (!$retval && !$exception) {
+                    return; // shutdown
+                }
+
+                /** @var \Psr\Http\Message\ServerRequestInterface $retval */
+                $activeSpan = \DDTrace\start_trace_span();
+                $activeSpan->service = \ddtrace_config_app_name('roadrunner');
+                $activeSpan->name = "web.request";
```
Comment (morrisonlevi) [inline src/Integrations/Integrations/Roadrunner/RoadrunnerIntegration.php:62]: @labbati This is a new integration. How confident are you in `http.server`? Should we set it to `web.request` for now?

File: tests/Integrations/Roadrunner/V2/CommonScenariosTest.php [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+                        'web.request',
+                        'roadrunner',
+                        'web',
+                        'GET /simple_view'
+                    )->withExactTags([
+                        'http.method' => 'GET',
+                        'http.url' => 'http://localhost:' . self::PORT . '/simple_view?key=value&<redacted>',
+                        'http.status_code' => '200',
+                    ]),
+                ],
```
Comment (morrisonlevi) [inline tests/Integrations/Roadrunner/V2/CommonScenariosTest.php:63]: I'm guessing roadrunner doesn't have a concept of a view?

File: tests/Unit/Integrations/IntegrationsLoaderTest.php [hunk 4]
```diff
@@ -175,6 +175,7 @@ public function testWeDidNotForgetToRegisterALibraryForAutoLoading()
             $excluded[] = 'symfony';
             $excluded[] = 'wordpress';
             $excluded[] = 'yii';
+            $excluded[] = 'roadrunner';
```
Comment (morrisonlevi) [inline tests/Unit/Integrations/IntegrationsLoaderTest.php:178]: This list appears to be in alphabetical order; can you insert it in the right place?

File: src/Integrations/Integrations/Roadrunner/RoadrunnerIntegration.php (hunk 1 above)
Comment (bwoebi) [reply src/Integrations/Integrations/Roadrunner/RoadrunnerIntegration.php:45]: No, we have a handler on the DD_TRACE_GENERATE_ROOT_SPAN config, which removes an existing root span if that's the top-most span.

File: src/Integrations/Integrations/Roadrunner/RoadrunnerIntegration.php (hunk 2 above)
Comment (bwoebi) [reply src/Integrations/Integrations/Roadrunner/RoadrunnerIntegration.php:62]: I would just keep it consistent, until we change them all.

File: tests/Integrations/Roadrunner/V2/CommonScenariosTest.php (hunk 3 above)
Comment (bwoebi) [reply tests/Integrations/Roadrunner/V2/CommonScenariosTest.php:63]: Yeah, it's just request/response model.

**Review by morrisonlevi — APPROVED** [review]

**Pull request merged**